v2/spanner-common/src/main/java/com/google/cloud/teleport/v2/spanner/migrations/shard/CassandraShard.java

```diff
@@ -0,0 +1,152 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.google.cloud.teleport.v2.spanner.migrations.shard;
+
+import com.datastax.oss.driver.api.core.config.DefaultDriverOption;
+import com.datastax.oss.driver.api.core.config.DriverConfigLoader;
+import com.datastax.oss.driver.api.core.config.DriverExecutionProfile;
+import java.util.List;
+import java.util.Objects;
+
+public class CassandraShard extends Shard {
+  private final DriverConfigLoader configLoader;
+
+  public CassandraShard(DriverConfigLoader configLoader) {
+    super();
+    this.configLoader = configLoader;
+    validateFields();
+    extractAndSetHostAndPort();
+  }
+
+  private void validateFields() {
+    if (getContactPoints() == null || getContactPoints().isEmpty()) {
+      throw new IllegalArgumentException("CONTACT_POINTS cannot be null or empty.");
+    }
+    if (getKeySpaceName() == null || getKeySpaceName().isEmpty()) {
+      throw new IllegalArgumentException("SESSION_KEYSPACE cannot be null or empty.");
+    }
+  }
+
+  private void extractAndSetHostAndPort() {
+    String firstContactPoint = getContactPoints().get(0);
+    String[] parts = firstContactPoint.split(":");
+
+    if (parts.length < 2) {
+      throw new IllegalArgumentException("Invalid contact point format: " + firstContactPoint);
+    }
+
+    String host = parts[0];
+    String port = parts[1];
+
+    setHost(host);
+    setPort(port);
+  }
+
+  private DriverExecutionProfile getProfile() {
+    return configLoader.getInitialConfig().getDefaultProfile();
+  }
+
+  // Getters that fetch data from DriverConfigLoader
+  public List<String> getContactPoints() {
+    return getProfile().getStringList(DefaultDriverOption.CONTACT_POINTS);
+  }
+
+  public String getKeySpaceName() {
+    return getProfile().getString(DefaultDriverOption.SESSION_KEYSPACE);
+  }
+
+  public String getConsistencyLevel() {
+    return getProfile().getString(DefaultDriverOption.REQUEST_CONSISTENCY, "LOCAL_QUORUM");
+  }
+
+  public boolean isSslEnabled() {
+    return getProfile().getBoolean(DefaultDriverOption.SSL_ENGINE_FACTORY_CLASS, false);
+  }
+
+  public String getProtocolVersion() {
+    return getProfile().getString(DefaultDriverOption.PROTOCOL_VERSION, "V5");
+  }
+
+  public String getDataCenter() {
+    return getProfile()
+        .getString(DefaultDriverOption.LOAD_BALANCING_LOCAL_DATACENTER, "datacenter1");
+  }
+
+  public int getLocalPoolSize() {
+    return getProfile().getInt(DefaultDriverOption.CONNECTION_POOL_LOCAL_SIZE, 1024);
+  }
+
+  public int getRemotePoolSize() {
+    return getProfile().getInt(DefaultDriverOption.CONNECTION_POOL_REMOTE_SIZE, 256);
+  }
+
+  public DriverConfigLoader getConfigLoader() {
```

Comment:
Except this, we probably don't need any of the other getters as ConfigLoader can directly build the session.

Reply:
You could use something on the line of 
```java
builder = CassandraDriverConfigLoader.fromOptionsMap(optionsMap). programmaticBuilder().
/* Override all your defaults here if unspecified in the builder*/
return builder.build()

```

Reply:
@VardhanThigle we are not building here , we may need this in our Connection helper

Reply:
@VardhanThigle we also need some of the getter under sourcewriterfn for handling mapping of connection and urls

Reply:
```
we also need some of the getter under sourcewriterfn for handling mapping of connection and urls
```
The minimum getters needed for the source Key thing can be retained.

Do we need `getProfile`,  `getUserName`, `getPassword` for generating mapping key as well?

```
we are not building here , we may need this in our Connection helper
```
This was not about building the session, it's about incorporating the defaults that you need to set.


The main point I would like to consider is the possibility that this layer of code should return the `DriverConfigLoader` (which is loaded from the file), and with any defaults if you want to set in case the file does not specify like quorum, added to it.

The connectionHelper, should just build the seession from the `DriverConfigLoader` returned from this layer.

Reply:
Other than the getter for mapping connection and urls, I don't see any other getter being used anywhere even in the Connection Helper

Reply:
Yes @shreyakhajanchi we have removed other getters already